airflow/jobs/scheduler_job.py

```diff
@@ -152,6 +152,22 @@ def __init__(
         self._zombie_threshold_secs = conf.getint("scheduler", "scheduler_zombie_task_threshold")
         self._standalone_dag_processor = conf.getboolean("scheduler", "standalone_dag_processor")
         self._dag_stale_not_seen_duration = conf.getint("scheduler", "dag_stale_not_seen_duration")
+
+        stalled_task_timeout = conf.getfloat("celery", "stalled_task_timeout", fallback=0)
+        if stalled_task_timeout:
+            # TODO: Remove in Airflow 3.0
+            warnings.warn(
+                "The 'celery.stalled_task_timeout' parameter is deprecated. "
+                "Please use 'scheduler.task_queued_timeout'.",
+                RemovedInAirflow3Warning,
```

Comment:
This is not the way to handle config deprecation. See https://github.com/apache/airflow/blob/1b4c6bed94580eab656cc476e99a4980813b11ca/airflow/configuration.py#L175-L180